actors/migration/power.go

```diff
@@ -101,19 +101,33 @@ func (m *powerMigrator) migrateCronEvents(ctx context.Context, store cbor.IpldSt
 	return migrateHAMTRaw(ctx, store, root)
 }
 
-func (m *powerMigrator) updateClaims(ctx context.Context, store cbor.IpldStore, root cid.Cid, updates *PowerUpdates) (cid.Cid, error) {
+func (m *powerMigrator) updateClaims(ctx context.Context, store cbor.IpldStore, root cid.Cid, updates *PowerUpdates, st *power0.State) error {
 	claims, err := adt0.AsMap(adt0.WrapStore(ctx, store), root)
 	if err != nil {
-		return cid.Undef, err
+		return err
 	}
 
 	for addr, claim := range updates.claims { // nolint:nomaprange
-		if err := claims.Put(abi.AddrKey(addr), &claim); err != nil {
-			return cid.Undef, err
+		rawPower := claim.RawBytePower
+		qaPower := claim.QualityAdjPower
+
+		var oldClaim power0.Claim
+		found, err := claims.Get(abi.AddrKey(addr), &oldClaim)
+		if err != nil {
+			return nil
+		}
+
+		if found {
+			rawPower = big.Sub(rawPower, oldClaim.RawBytePower)
+			qaPower = big.Sub(qaPower, oldClaim.QualityAdjPower)
+		}
+
+		if err := st.AddToClaim(adt0.WrapStore(ctx, store), addr, rawPower, qaPower); err != nil {
+			return err
```

Comment:
See my comment on #1237, but in any case don't include this fix in this otherwise validation-only PR.

Reply:
reverted